src/objects/Mesh.js

```diff
@@ -143,213 +147,244 @@ Mesh.prototype = Object.assign( Object.create( Object3D.prototype ), {
 
 		if ( material === undefined ) return;
 
-		// Checking boundingSphere distance to ray
+		//Determine how many times raycast will execute
```

Comment:
I've actually thought of something different^^. I would you try to use `Mesh.raycast()` in `InstancedMesh.raycast()` in some way. For example by using a temporary instance of `Mesh` to represents the single instances of `InstancedMesh`. In this way, you should be able to delegate the raycast to the super method without modifying it.

It's actually no good code style if superclasses know details of their subclasses. Hence, I'm not a fan of using this style in `Mesh.raycast()` by letting the method know about instancing.

Reply:
You're right. I'd like to ask instnacemesh.raycast to call the parent class mesh.raycast in some way, similar to super in Java, but I didn't expect to implement it in the way of a single instance of instancdmesh in the way of mesh temporary agent. This is really a good idea. I'll try to implement it again in this way or some other way, rather than the way it is now.